query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverter.java

```diff
@@ -33,25 +39,67 @@ public PinotFunctionConverter() {
     this.percentileAggFunction = DEFAULT_PERCENTILE_AGGREGATION_FUNCTION;
   }
 
-  public String convert(
-      Function function, java.util.function.Function<Expression, String> argumentConverter) {
+  public String convert(ExecutionContext executionContext, Function function, java.util.function.Function<Expression, String> argumentConverter) {
     switch (function.getFunctionName().toUpperCase()) {
       case QUERY_FUNCTION_COUNT:
         return this.convertCount();
       case QUERY_FUNCTION_PERCENTILE:
         return this.functionToString(this.toPinotPercentile(function), argumentConverter);
       case QUERY_FUNCTION_CONCAT:
         return this.functionToString(this.toPinotConcat(function), argumentConverter);
+      case QUERY_FUNCTION_AVG_RATE:
+        return this.functionToStringForAvgRate(function, argumentConverter, executionContext);
       default:
         // TODO remove once pinot-specific logic removed from gateway - this normalization reverts
         // that logic
         if (this.isHardcodedPercentile(function)) {
-          return this.convert(this.normalizeHardcodedPercentile(function), argumentConverter);
+          return this.convert(executionContext, this.normalizeHardcodedPercentile(function), argumentConverter);
         }
         return this.functionToString(function, argumentConverter);
     }
   }
 
+  private String functionToStringForAvgRate(Function function, java.util.function.Function<Expression, String> argumentConverter, ExecutionContext executionContext) {
+    function = updateFunctionForAvgRate(function, executionContext);
+    String argumentString =
+        function.getArgumentsList().stream()
+            .map(argumentConverter::apply)
+            .collect(Collectors.joining(","));
+
+    return "SUM(DIV(" + argumentString + "))";
+  }
+
+  private Function updateFunctionForAvgRate(Function function, ExecutionContext executionContext){
+
+    Expression columnName = function.getArgumentsList().get(0);
+    Expression literal = function.getArgumentsList().get(1);
+
+    Builder builder = function.toBuilder();
+    builder.clearArguments();
+    builder.addArguments(columnName);
+    builder.addArguments(getUpdatedLiteral(columnName, literal, executionContext));
+
+    return builder.build();
+  }
+
+  private Expression getUpdatedLiteral(Expression column, Expression literal, ExecutionContext executionContext){
+
+    String columnName = column.getColumnIdentifier().getColumnName().split("[.]")[0];
+    long divisorInSeconds = literal.getLiteral().getValue().getLong();
```

Comment:
Can we avoid the assumption that this will be in seconds? We normally use milliseconds for one, but we should require the units to be specified since this is a new arg anyway.

Reply:
We use Value to represent this so I'm not sure if we can do that with the current proto.

Reply:
We can transport is as a well known string for now - that's what we do for group by (I'd suggest an ISO duration string).

Reply:
CC @kotharironak

Reply:
Yes. Agree that we should support the duration with type in avgrate(column, "30:seconds"). But, I think, there will change requires from graphql to gateway layer too. B'case, I see following request at Gateway
```
selection {
  function {
    function: AVGRATE
    arguments {
      columnIdentifier {
        columnName: "BACKEND.errorCount"
      }
    }
    arguments {
      literal {
        value {
          valueType: LONG
          long: 1
        }
      }
    }
    alias: "AVGRATE_BACKEND.errorCount_[1]"
  }
}
```

But, that said, let's fix this. @sarthak77 As @aaron-steinfeld suggested, we should consider the argument of avgrate as a string in ISO duration form.

Reply:
Assuming time in time series request in ISO format. It caused couple of tests to fail so disabled them as of now until changes in other layers are done.

Reply:
You don't have to change the format of dateTimeConvert function as part of this ticket. I was suggesting using ISO format in avg_rate argument for rateInerval.